src/main/java/org/mifos/connector/ams/paygops/camel/route/PaygopsRouteBuilder.java

```diff
@@ -66,20 +66,31 @@ public void configure() {
                 .log(LoggingLevel.INFO, "Paygops Validation Response Received")
                 .process(exchange -> {
                     // processing success case
-                    String body = exchange.getIn().getBody(String.class);
-                    ObjectMapper mapper = new ObjectMapper();
-                    PaygopsResponseDto result = mapper.readValue(body, PaygopsResponseDto.class);
-                    logger.info("body : "+ result);
-                    //JSONObject jsonObject = new JSONObject(body);
-                    if (result.getReconciled()){
+                    try {
+                        String body = exchange.getIn().getBody(String.class);
+                        ObjectMapper mapper = new ObjectMapper();
+                        PaygopsResponseDto result = mapper.readValue(body, PaygopsResponseDto.class);
+                        logger.info("body : "+ result);
+                        //JSONObject jsonObject = new JSONObject(body);
+                        if (result.getReconciled()){
                             logger.info("Paygops Validation Successful");
                             exchange.setProperty(PARTY_LOOKUP_FAILED, false);
                         }
                         else {
-                        logger.info("Paygops Validation Unsuccessful, Reconciled field returned false");
+                            logger.info("Paygops Validation Unsuccessful, Reconciled field returned false");
+                            exchange.setProperty(ERROR_DESCRIPTION, result);
```

Comment:
setCamelErrorInfo has to be made into a function and reused as well.

Reply:
okay